cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/DFGPass.kt

```diff
@@ -90,6 +92,15 @@ class DFGPass : Pass() {
         }
     }
 
+    /**
+     * For a [DeclarationStatement], the whole statement flows into its declarations. This is fine
+     * because it's used as a wrapper if a Statement is needed but we only have a Declaration (which
+     * is not a statement).
+     */
+    private fun handleDeclarationStatement(node: DeclarationStatement) {
```

Comment:
Decision: Not handling this for *all* declaration statements, but only do this for the `ForStatement`.

Reply:
I removed this method and handle the `ForEachStatement` differently now. @maximiliankaul please check if the new handling makes sense to you